onnxruntime/test/providers/cpu/tensor/tensor_op_test.cc

```diff
@@ -162,6 +162,30 @@ TEST(TensorOpTest, Reshape_UnknownDimWithAllowZero) {
   test.Run();
 }
 
+TEST(TensorOpTest, ReshapeSixDimNewShape) {
```

Comment:
I'm on the fence about whether it is worth adding these 6d reshape/transpose tests